### PR TITLE
Add NVIDIA MIG support

### PR DESCRIPTION
**Pull request by stgraber**

### CODE REVIEW

File: lxd/instance/drivers/driver_lxc.go [hunk 1]
```diff
@@ -2112,6 +2113,19 @@ func (d *lxc) startCommon() (string, []func() error, error) {
 		if len(runConf.PostHooks) > 0 {
 			postStartHooks = append(postStartHooks, runConf.PostHooks...)
 		}
+
+		// Build list of NVIDIA GPUs (used for MIG).
+		if len(runConf.GPUDevice) > 0 {
```
Comment (tomponline) [inline lxd/instance/drivers/driver_lxc.go:2118]: @stgraber this isn't ideal as its exposing device (and vendor) specific logic outside of the driver, and is using (at least conceptually) a vendor-neutral config key (`runConf.GPUDevice`) for detection of vendor-specific actions (setting the nvidia env var).

I was considering adding support for an `Environment` field in the `runConfig` struct, but the requirement to concatenate the the names of the different devices together into a single env var prevents that.

Also, the use of `runConf.GPUDevice[0]` below is brittle, as there are no guarantees of the ordering of the `RunConfig` items, and so there is also the potential if we use `GPUDevice` for some other type of GPU config in the future to trip this part of the code up.

So to make this part at least less brittle and to clearly tie the population of `nvidiaDevices` to nvidia MIG cards in code, I suggest the following:

1. Add a constant in the `device/gpu_mig.go` file called something like `GPUNvidiaDeviceKey = "nvidia.device"`.
2. Use this constant as the key when populating the GPUDevice config in line https://github.com/lxc/lxd/pull/8627/files#diff-3bfc08351aba94ff0724401b7d5a57c97f51de8c559ad36107c7d2f114d3ebd1R125
3. Then here, iterate through the `runConf.GPUDevice` items looking for one that has a `Key` field that equals `device.GPUNvidiaDeviceKey` value, then use that to append into the `nvidiaDevices` slice.

That will fix the ordering issue, prevent accidental non-nvidia GPU configs from being used here in the future, and provide an easy way to identify the users of the "nvidia.device" config key via use of the constant.

Comment (tomponline) [reply lxd/instance/drivers/driver_lxc.go:2118]: @stgraber if you need this merged as-is then we can look at these tweaks later though.

**Comment by stgraber** [comment]
@tomponline feedback addressed

**Comment by tomponline** [comment]
@stgraber thanks. The constant needs a comment to appease static analysis then good to merge. 😀

**Comment by stgraber** [comment]
@tomponline added now :)

**Comment by tomponline** [comment]
Ah just saw you added it just now

**Pull request merged**